apps/mysql.go

```diff
@@ -15,13 +15,75 @@
 package apps
 
 import (
-	"fmt"
-	"strings"
+	"os"
 
 	"github.com/GoogleCloudPlatform/ops-agent/confgenerator"
 	"github.com/GoogleCloudPlatform/ops-agent/confgenerator/fluentbit"
+	"github.com/GoogleCloudPlatform/ops-agent/confgenerator/otel"
+
+	"fmt"
+	"strings"
 )
 
+type MetricsReceiverMySQL struct {
+	confgenerator.ConfigComponent `yaml:",inline"`
+
+	confgenerator.MetricsReceiverShared `yaml:",inline"`
+
+	Endpoint string `yaml:"endpoint" validate:"omitempty,hostname_port|file"`
+
+	Password  string `yaml:"password" validate:"omitempty"`
+	Username  string `yaml:"username" validate:"omitempty"`
```

Comment:
We don't seem to have a default for username; does an empty username get turned into `root` somewhere downstream?

Reply:
it does now.

Reply:
Looks like username and password are no longe required. https://github.com/GoogleCloudPlatform/ops-agent/pull/218/commits/4a4a8195aa05fd30f5b6f598783022e4307fcaee and https://github.com/GoogleCloudPlatform/ops-agent/pull/218/commits/d9f213f6764b6dca1e2ea9f456c31be1a089657e also removed the related username and password settings. Resolving this.